src/frontend/contexts/LocalDiscoveryContext.tsx

```diff
@@ -91,33 +98,48 @@ export function createLocalDiscoveryController(mapeoApi: MapeoClientApi) {
   };
   let cancelNetInfoFetch: undefined | (() => void);
   const zeroconf = new Zeroconf();
+  // In edge-cases, we may end up with multiple published names (NSD service
+  // will append ` (1)` to the name if there is a conflict), so we need to track
+  // them here so that we can unpublish them when we stop the service.
+  const publishedNames = new Set<string>();
 
   const sm = new StateMachine({
     async start() {
-      const [{name, port}] = await Promise.all([
-        mapeoApi.startLocalPeerDiscoveryServer(),
-        startZeroconf(zeroconf),
-      ]);
-      zeroconf.publishService(
-        ZEROCONF_SERVICE_TYPE,
-        ZEROCONF_PROTOCOL,
-        ZEROCONF_DOMAIN,
-        name,
-        port,
+      // start browsing straight away
+      const startZeroconfPromise = startZeroconf(zeroconf);
+      const {name, port} = await mapeoApi.startLocalPeerDiscoveryServer();
+      // publishedName could be different from the name we requested, if there
+      // was a conflict on the network (the conflict could come from the same
+      // name still being registered on the network and not yet cleaned up)
+      const publishedName = await publishZeroconf(zeroconf, {name, port}).catch(
+        e => {
+          // Publishing could fail (timeout), but we don't want to throw the
+          // state machine start(), because that would leave the state machine
+          // in an "error" state and stop other things from working. By silently
+          // failing (with the report to Sentry), we are able to try again next
+          // time.
+          Sentry.captureException(e);
+        },
       );
+      if (publishedName) publishedNames.add(publishedName);
```

Comment:
I found this last conditional a little confusing, because it implies that `publishZeroconf` can return a falsy value. I think using a normal `try`/`catch` feels a bit clearer:

```suggestion
      try {
        const publishedName = await publishZeroconf(zeroconf, {name, port});
        publishedNames.add(publishedName);
      } catch (e) {
        // Publishing could fail (timeout), but we don't want to throw the
        // state machine start(), because that would leave the state machine
        // in an "error" state and stop other things from working. By silently
        // failing (with the report to Sentry), we are able to try again next
        // time.
        Sentry.captureException(e);
      }
```

Reply:
Agree to this suggestion. I've used this pattern (adding catch() and testing return value) a couple of times and it's probably not a great pattern!